python/cudf/cudf/tests/test_copying.py

```diff
@@ -52,3 +53,324 @@ def test_null_copy():
+    for ser in [s, s2, s3, s4, s5, s6]:
+        assert_eq(ser, cudf.Series([10, 20, 30, 40, 50]))
+
+    s6[0:3] = 3000
+    assert_eq(s1, cudf.Series([10000, 10000, 10000, 40, 50]))
+    assert_eq(s6, cudf.Series([3000, 3000, 3000, 40, 50]))
+    for ser in [s2, s3, s4, s5]:
+        assert_eq(ser, cudf.Series([10, 20, 30, 40, 50]))
+
+    s2[1:4] = 4000
+    assert_eq(s2, cudf.Series([10, 4000, 4000, 4000, 50]))
+    assert_eq(s1, cudf.Series([10000, 10000, 10000, 40, 50]))
+    assert_eq(s6, cudf.Series([3000, 3000, 3000, 40, 50]))
+    for ser in [s3, s4, s5]:
+        assert_eq(ser, cudf.Series([10, 20, 30, 40, 50]))
+
+    s4[2:4] = 5000
+    assert_eq(s4, cudf.Series([10, 20, 5000, 5000, 50]))
+    assert_eq(s2, cudf.Series([10, 4000, 4000, 4000, 50]))
+    assert_eq(s1, cudf.Series([10000, 10000, 10000, 40, 50]))
+    assert_eq(s6, cudf.Series([3000, 3000, 3000, 40, 50]))
+    for ser in [s3, s5]:
+        assert_eq(ser, cudf.Series([10, 20, 30, 40, 50]))
+
+    s5[2:4] = 6000
+    assert_eq(s5, cudf.Series([10, 20, 6000, 6000, 50]))
+    assert_eq(s4, cudf.Series([10, 20, 5000, 5000, 50]))
+    assert_eq(s2, cudf.Series([10, 4000, 4000, 4000, 50]))
+    assert_eq(s1, cudf.Series([10000, 10000, 10000, 40, 50]))
+    assert_eq(s6, cudf.Series([3000, 3000, 3000, 40, 50]))
+    for ser in [s3]:
+        assert_eq(ser, cudf.Series([10, 20, 30, 40, 50]))
+
+    s7 = s5.copy(deep=False)
+    assert_eq(s7, cudf.Series([10, 20, 6000, 6000, 50]))
+    s7[1:3] = 55
+    assert_eq(s7, cudf.Series([10, 55, 55, 6000, 50]))
+
+    assert_eq(s4, cudf.Series([10, 20, 5000, 5000, 50]))
+    assert_eq(s2, cudf.Series([10, 4000, 4000, 4000, 50]))
+    assert_eq(s1, cudf.Series([10000, 10000, 10000, 40, 50]))
+    assert_eq(s6, cudf.Series([3000, 3000, 3000, 40, 50]))
+    for ser in [s3]:
+        assert_eq(ser, cudf.Series([10, 20, 30, 40, 50]))
+
+    del s2
+
+    assert_eq(s1, cudf.Series([10000, 10000, 10000, 40, 50]))
+    assert_eq(s3, cudf.Series([10, 20, 30, 40, 50]))
+    assert_eq(s4, cudf.Series([10, 20, 5000, 5000, 50]))
+    assert_eq(s5, cudf.Series([10, 20, 6000, 6000, 50]))
+    assert_eq(s6, cudf.Series([3000, 3000, 3000, 40, 50]))
+    assert_eq(s7, cudf.Series([10, 55, 55, 6000, 50]))
+
+    del s4
+    del s1
+
+    assert_eq(s3, cudf.Series([10, 20, 30, 40, 50]))
+    assert_eq(s5, cudf.Series([10, 20, 6000, 6000, 50]))
+    assert_eq(s6, cudf.Series([3000, 3000, 3000, 40, 50]))
+    assert_eq(s7, cudf.Series([10, 55, 55, 6000, 50]))
+
+    del s
+    del s6
+
+    assert_eq(s3, cudf.Series([10, 20, 30, 40, 50]))
+    assert_eq(s5, cudf.Series([10, 20, 6000, 6000, 50]))
+    assert_eq(s7, cudf.Series([10, 55, 55, 6000, 50]))
+
+    del s5
+
+    assert_eq(s3, cudf.Series([10, 20, 30, 40, 50]))
+    assert_eq(s7, cudf.Series([10, 55, 55, 6000, 50]))
+
+    del s3
+    assert_eq(s7, cudf.Series([10, 55, 55, 6000, 50]))
+    cudf.set_option("copy_on_write", original_cow_setting)
+
+
+def test_series_zero_copy_cow_on():
```

Comment:
Same as above, can you add a comment explaining the purpose of this test.

Reply:
Done